### PR TITLE
feat(ci): cross-compile release assets, cache deps

### PR DESCRIPTION
**Pull request by rymnc**
Updates CI to include -
- Caching for dependencies
- Use `make installdeps` instead of manually installing submodules
- Use `cross-rs` to cross-compile for different targets

### CODE REVIEW

**Comment by rymnc** [comment]
Tested the cross-compiled assets locally ~ works on apple silicon macs now

File: Makefile [hunk 1]
```diff
@@ -6,9 +6,8 @@ all: .pre-build build
 	@git submodule update --init --recursive
 
 .pre-build: .fetch-submodules
-ifeq (, $(shell which cargo-make))
-	@cargo install --force cargo-make
-endif
+	@cargo install cargo-make
+	@cargo install cross --git https://github.com/cross-rs/cross --branch main
```
[the remark this answers is not in the document]
Comment (rymnc) [reply Makefile:10]: Cross is just used for cross-compiling, imo general users shouldn't have to use it, but  users wishing to cross-compile would have prior knowledge of it. wdyt?

[the remark this answers is not in the document]
Comment (rymnc) [reply Makefile:10]: How about this?
in the makefile-
```makefile
installdeps:
    @cargo install cargo-make # as usual
ifdef CI
    @cargo install cross ...
endif
```

[the remark this answers is not in the document]
Comment (rymnc) [reply Makefile:11]: Addressed in [2629c54](https://github.com/vacp2p/zerokit/pull/128/commits/2629c54850b04d396cfea86e1e7671d356c542ab)

**Review by tyshko-rostyslav — APPROVED** [review]
LGTM!

**Pull request merged**